projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -461,18 +482,51 @@ async def run(self) -> int:
         res = 0
         loops = 0
 
-        async with await psycopg.AsyncConnection.connect(self.db_url) as conn:
+        async with await psycopg.AsyncConnection.connect(
+            self.db_url, autocommit=True
+        ) as conn:
             await register_vector_async(conn)
             await self.vectorizer.config.embedding.setup()
             while True:
-                if not self._continue_processing(loops, res):
+                if not await self._continue_processing(conn, loops, res):
                     return res
                 items_processed = await self._do_batch(conn)
                 if items_processed == 0:
                     return res
                 res += items_processed
                 loops += 1
 
+    async def _continue_processing(
```

Comment:
Nit: by reading the func name, seems you are telling and not asking. In other words, it seems the intention is to actually resume processing instead of asking if the process should continue.